tests/sample/trainers/test_forward_dynamics.py

```diff
@@ -0,0 +1,149 @@
+from sample.models.components.deterministic_normal import FCDeterministicNormalHead
+from sample.models.components.qlstm import QLSTM
+from sample.models.forward_dynamics import ForwardDynamics
+from sample.trainers.forward_dynamics import ForwardDynamicsTrainer
+from sample.trainers.sampler import RandomTimeSeriesSampler
+from tests.sample.helpers import parametrize_device
+
+
+class TestForwardDynamicsTrainer:
+    BATCH = 4
+    DEPTH = 8
+    DIM = 16
+    DIM_FF_HIDDEN = 32
+    LEN = 64
+    LEN_SEQ = 16
+    DROPOUT = 0.1
+    DIM_OBS = 32
+    DIM_ACTION = 8
+
+    @pytest.fixture
+    def core_model(self):
+        qlstm = QLSTM(self.DEPTH, self.DIM, self.DIM_FF_HIDDEN, self.DROPOUT)
+        return qlstm
+
+    @pytest.fixture
+    def observation_flatten(self):
+        return nn.Identity()
+
+    @pytest.fixture
+    def action_flatten(self):
+        return nn.Identity()
+
+    @pytest.fixture
+    def obs_action_projection(self):
+        return nn.Linear(self.DIM_OBS + self.DIM_ACTION, self.DIM)
+
+    @pytest.fixture
+    def obs_hat_dist_head(self):
+        return FCDeterministicNormalHead(self.DIM, self.DIM_OBS)
+
+    @pytest.fixture
+    def forward_dynamics(
+        self,
+        observation_flatten,
+        action_flatten,
+        obs_action_projection,
+        core_model,
+        obs_hat_dist_head,
+    ):
+        return ForwardDynamics(
+            observation_flatten,
+            action_flatten,
+            obs_action_projection,
+            core_model,
+            obs_hat_dist_head,
+        )
+
+    @pytest.fixture
+    def data_buffers(self):
+        return {
+            BufferName.FORWARD_DYNAMICS: SequentialBuffer(
+                [DataKey.OBSERVATION, DataKey.ACTION, DataKey.HIDDEN], max_size=self.LEN
+            )
+        }
+
+    @pytest.fixture
+    def partial_dataloader(self):
+        partial_dataloader = partial(DataLoader, batch_size=2)
+        return partial_dataloader
+
+    @pytest.fixture
+    def partial_sampler(self):
+        partial_sampler = partial(RandomTimeSeriesSampler, sequence_length=self.LEN_SEQ)
+        return partial_sampler
+
+    @pytest.fixture
+    def partial_optimizer(self):
+        partial_optimizer = partial(AdamW, lr=1e-4, weight_decay=0.04)
+        return partial_optimizer
+
+    @pytest.fixture
+    def trainer(
+        self,
+        partial_dataloader,
+        partial_sampler,
+        partial_optimizer,
+        mocker: MockerFixture,
+    ):
+        mocker.patch("sample.trainers.forward_dynamics.mlflow")
+        trainer = ForwardDynamicsTrainer(
+            partial_dataloader,
+            partial_sampler,
+            partial_optimizer,
+            min_buffer_size=4,
+            min_new_data_count=2,
+        )
+        return trainer
+
+    @parametrize_device
+    def test_run(self, device, data_buffers, forward_dynamics, trainer):
+        models = {
+            ModelName.FORWARD_DYNAMICS: TorchTrainingModel(
```

Comment:
`StrEnum`は`str`のサブクラスではあるのですが、辞書オブジェクトではキーの型は不変なので、`str`にキャストしないといけないようです。

どういった問題が起きるかというと、`StrEnum`の要素の `VALUE = "value"`と ただの文字列の `"value"`はあくまでも異なるオブジェクトであるが、辞書オブジェクトからすれば区別がつかなくなってしまう、とのことだそうです。私も少しまだ理解できてないのですが、 StrEnumを完全に`str`と同じように扱うことはできないそうな。

```suggestion
            str(ModelName.FORWARD_DYNAMICS): TorchTrainingModel(
```